joomla/joomla-cms
# [5.0] Custom Error Pages - new feature

### Summary of Changes
Adds a module position to the error.php page of cassiopeia so that you can customise the 404 page without having to change language strings or asking the user to modify the error.php (which they will lose on update)

This is fully backwards compatible as the existing error page text is displayed if there is no module. If debug is enabled then the debug information is still displayed

In theory the error.php is used by joomla for all error codes so this PR is able to display a specific module for each error code.


### Testing Instructions
Apply the PR
Create a new module (custom html is probably the most useful) and publish in the new position error-404
Go to a fake page on your site and you will see the 404 page with your new module
Unpublish the module and repeat and you will see the original 404 page with the language strings


### Original 404
![image](https://github.com/joomla/joomla-cms/assets/1296369/fb576ca9-7824-4cbf-9ecf-d302b9efc96c)




### Example 404 AFTER applying this Pull Request and creating a module

![image](https://github.com/joomla/joomla-cms/assets/1296369/c6c915fe-437f-4abd-9870-c07c12ec95cb)


![image](https://github.com/joomla/joomla-cms/assets/1296369/9d7a083c-6fd8-4927-9e17-3267c85bb087)


If accepted then I will create a guided tour for this and post-installation message

### Link to documentations
Please select:


cc @crystalenka @HLeithner @alexandreelise
